handlers/changes.js

```diff
@@ -129,105 +146,107 @@ var getChanges = function(feed) {
   });
 };
 
+var bindServerIds = function(feed, callback) {
+  bindSubjectIds(feed, function(err) {
+    if (err) {
+      return callback(err);
+    }
+    bindValidatedDocIds(feed, callback);
+  });
+};
+
 var initFeed = function(feed, callback) {
   bindRequestedIds(feed, function(err) {
     if (err) {
       return callback(err);
     }
-    bindViewKeys(feed, function(err) {
-      if (err) {
-        return callback(err);
-      }
-      bindValidatedDocIds(feed, callback);
-    });
+    bindServerIds(feed, callback);
   });
 };
 
 // returns if it is true that for any document in the feed the user
 // should be able to see it AND they don't already
-var hasNewApplicableDoc = function(feed, docs) {
-  return _.some(docs, function(doc) {
-    return !_.contains(feed.validatedIds, doc.id) &&
-      _.some(doc.keys, function(key) {
-        return !!_.find(feed.keys, function(feedKey) {
-          return feedKey[0] === key[0];
-        });
-      });
-  });
-};
-
-// WARNING: If updating this function also update the doc_by_place view in lib/views.js
-var extractKeysFromDoc = function(doc, emit) {
-
-  var emitPlace = function(place) {
-    var depth = 0;
-    while (place) {
-      if (place._id) {
-        emit([ place._id ]);
-        emit([ place._id, depth ]);
+var hasNewApplicableDoc = function(feed, changes) {
+  return _.some(changes, function(change) {
+    if (_.contains(feed.validatedIds, change.id)) {
+      // feed already knows about doc
+      return false;
+    }
+    if (feed.subjectIds.indexOf(change.subjectId) !== -1) {
+      // this is relevant to the feed
+      return true;
+    }
+    if (CONTACT_TYPES.indexOf(change.doc.type) === -1) {
+      // not a contact type so can't be a new subject
```

Comment:
Because subjects are people and places never data_records. I'll update the comment.